src/main/java/com/codeborne/selenide/impl/CiReportUrl.java

```diff
@@ -0,0 +1,54 @@
+package com.codeborne.selenide.impl;
+
+import com.codeborne.selenide.SelenideConfig;
+
+import java.util.logging.Logger;
+
+public class CiReportUrl {
+  private static final Logger LOG = Logger.getLogger(SelenideConfig.class.getName());
+
+  public String getReportsUrl(String reportsUrl) {
+    if (!isEmpty(reportsUrl)) {
+      LOG.config("Using variable selenide.reportsUrl=" + reportsUrl);
+      return reportsUrl;
+    }
+    reportsUrl = getJenkinsReportsUrl();
+    if (!isEmpty(reportsUrl)) {
+      LOG.config("Using Jenkins BUILD_URL: " + reportsUrl);
+      return reportsUrl;
+    }
+    reportsUrl = getTeamCityUrl();
+    if (!isEmpty(reportsUrl)) {
+      LOG.config("Not found one of [teamcity.serverUrl,teamcity.buildType.id,build.number]. It's not Teamcity.");
+      return reportsUrl;
+    }
+    LOG.config("Variable selenide.reportsUrl not found");
+    return reportsUrl;
+  }
+
+  private String getTeamCityUrl() {
+    String url = System.getProperty("teamcity.serverUrl");
+    String build_type = System.getProperty("teamcity.buildType.id");
+    String build_number = System.getProperty("build.number");
+    String result = url + "/repository/download/" + build_type + "/" + build_number + ":id/";
```

Comment:
Let's use `Paths.get(url, "repository", "download", build_type, ...).toString()` instead of summ operation.
In this case we don't case about platform's folders/url dividers.
And avoid `/` symbol duplication.

Reply:
We can get illegal characters from the reportUrl like ':id', Paths solution not fit for here. We should use url builders and etc. but it is overhead, hence added url format checking and normalization (for remove double slashes)